Movement_Handler.py

```diff
@@ -51,54 +62,71 @@ def set_parking_slot_destination(self):
-            nearest_parking_slot_path = self.BFS_Logic.shortestPath(Detection_controller.get_matrix(), self.car.get_position(), self.slot)
+            nearest_parking_slot_path = self.BFS_Logic.shortestPath(Detection_controller.get_matrix(),
+                                                                    self.car.get_position(), self.slot)
             if len(nearest_parking_slot_path) < dist_nearest_parking_slot:
                 dist_nearest_parking_slot = len(nearest_parking_slot_path)
                 chosen_slot = slot
 
         self.parking_slot_dest = chosen_slot
 
-
     def car_movement(self, matrix):
         path = self.BFS_Logic.shortestPath(Detection_controller.get_matrix(), self.car.get_position(),
                                            self.parking_slot_dest)
-        # understand the next cell
-        # update car angle commands needed
-        # move to the right angle
-        # move forward for the next cell (maybe we will decide that can move few steps forward)
+        if path is None:
+            print("No valid path found.")
+            return
+
+        for i in range(len(path) - 1):
+            current_cell = path[i]
+            next_cell = path[i + 1]
 
+            next_direction = self.get_next_direction(current_cell, next_cell)
+            update_car_angle(self.car, angle_to_direction.get(next_direction))
 
+            # Move the car forward (you need to implement this part)
+            # TODO: Understand how much forward need to move!?!?
+            self.move_car_forward()
 
-    def update_car_angle(self, car, next_direction):
+    @staticmethod
+    def update_car_angle(car, next_direction):
         """
         :param car: object car for getting current angle of car
         :param next_direction: next direction in degrees (if right needed, it would be 180)
         :return: move command value (left/right) and num of steps that need to be done
         """
-        num_of_degrees = abs(car.direction - next_direction)
+        car_tilt_degrees = car.get_direction_degrees()  # 315 --> 0
+        num_of_degrees = abs(car_tilt_degrees - next_direction)  # =315
         direction = None
         if num_of_degrees > 180:
-            num_of_steps = (int)((360 - num_of_degrees)/RIGHT_LEFT_DEGREE)
-            direction = MOVE_COMMANDS.Left
+            num_of_steps = (int)((360 - num_of_degrees) / RIGHT_LEFT_DEGREE)
+            direction = MOVE_COMMANDS.RIGHT
         else:
-            num_of_steps = (int)((num_of_degrees)/RIGHT_LEFT_DEGREE)
-            direction = MOVE_COMMANDS.Right
+            num_of_steps = (int)((num_of_degrees) / RIGHT_LEFT_DEGREE)
+            direction = MOVE_COMMANDS.LEFT
 
-        return num_of_steps, direction
+        move_to_correct_angle(num_of_steps, direction)
 
+    @staticmethod
+    def move_to_correct_angle(num_of_moves: int, direction: MOVE_COMMANDS):
+        for i in range(num_of_moves):
+            move(angle_to_direction)
```

Comment:
move needs to get a direction (MOVE_COMMANDS) and not 'angle_to_direction'. angle_to_direction is a dictionary that gets a direction and returns a number but the move function gets only (Forward/Back/Left/Right)